src/bosh-director/lib/bosh/director/jobs/ssh.rb

```diff
@@ -21,17 +22,32 @@ def initialize(deployment_id, options = {})
 
       def perform
         target = Target.new(@target_payload)
-
         filter = {}
-        filter[:job] = target.job if target.job
-        filter.merge!(target.id_filter)
+        if target.job && ip_address?(target.job)
```

Comment:
Theoretically, I suppose there could be ambiguity on whether it's a job or an IP address. Is it worth the complexity of only falling back to IP address if job name did not match? Probably not, but I wondered if you had considered the possibility.

Reply:
@dpb587-pivotal at this moment you cannot specify job name as something like '1.1.1.1', because of error `Invalid DNS canonical name '...', must begin with a letter`. At least that's what I had. But you are correct, it's better to cover this possibility for the future possible updates.